R/desc_utils.R

```diff
@@ -0,0 +1,201 @@
+  desc_field_pkg_names <- map_key_character(desc_field_pkgs, "package")
+
+  # Parse the remotes to retrieve the package names
+  remotes <- pkgdepends::parse_pkg_refs(d$get_remotes())
+  remotes_pkg <- map_key_character(remotes, "package")
+
+  # Add to remotes `Config/Needs/verdepcheck` that resolve to a remote_ref_github
+  desc_field_include_ix <- vapply(desc_field_pkgs, inherits, logical(1), "remote_ref_github")
+
+  # Only keep previous remotes that are not defined in `Config/Needs/verdepcheck`
+  remotes_include_ix <- remotes_pkg %in% setdiff(remotes_pkg, desc_field_pkg_names)
+
+  # Create new list of references that will be used as "Remotes"
+  new_remotes <- c(
+    map_key_character(desc_field_pkgs[desc_field_include_ix], "ref"),
+    map_key_character(remotes[remotes_include_ix], "ref")
+  )
+
+  # Remove all remotes and override it
+  d$clear_remotes()
+
+  # Return clause without Remotes section if none should be kept
+  if (is.null(new_remotes) || length(new_remotes) == 0) {
+    return(d)
+  }
+  d$set_remotes(new_remotes)
+  d
+}
+
+#' Set `"Config/Needs/verdepcheck"` section into the `desc` object if not empty else clear this section.
+#' @keywords internal
+desc_cond_set_refs <- function(d, refs) {
+  if (length(refs)) {
+    d$set_list(.desc_field, refs)
+  } else {
+    d$del(.desc_field)
+  }
+  return(invisible(d))
+}
+
+#' Create `installation_plan` object from `desc` object
+#' @importFrom pkgdepends new_pkg_deps new_pkg_installation_proposal
+#' @keywords internal
+desc_to_ip <- function(d, config) {
+  temp_desc <- tempfile()
+  d$write(temp_desc)
+
+  pkgdepends::new_pkg_installation_proposal(
+    refs = paste0("deps::", temp_desc),
+    config = config
+  )
+}
+
+#' Get package version from description
+#'
+#' @keywords internal
+#'
+#' @examples
+#' d <- desc::desc(cmd = "!new")
+#'
+#' d$set_dep("magrittr", type = "Imports", version = "*")
+#' verdepcheck:::version_from_desc("magrittr", d)
+#'
+#' d$set_dep("magrittr", type = "Imports", version = ">= 1.5")
+#' verdepcheck:::version_from_desc("magrittr", d)
+version_from_desc <- function(pkg_name, desc) {
+  subset_versions <- function(.x) (.x$version[.x$package == pkg_name])[[1]]
+  version <- subset_versions(desc$get_deps())
+  result <- list(
+    package = pkg_name,
+    version_str = version,
+    op = "",
+    op_ver = ""
+  )
+  if (version == "*" || trimws(version) == "") {
+    return(result)
+  }
+  split_vec <- strsplit(version, " ")[[1]]
+  result$version_str <- version
+  result$op <- split_vec[1]
+  result$op_ver <- split_vec[2]
+  result
+}
+
+#' Filter for package versions that comply with an operator and version
+#'
+#' @param x `vector` of valid package versions.
+#' @param op `character(1)` relational operator (`>=`, `==`, ...)
+#' @param op_ver `character(1)` or `package_version(1)` with version to compare
+#' with using a relational operator.
+#'
+#' @keywords internal
+#'
+#' @examples
+#' versions <- paste(1:10, rep("0", 10), sep = ".")
```

Comment:
a little bit simpler

```suggestion
#' versions <- paste(1:10, 0, sep = ".")
```